jenkins/osx_c.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+# Copyright (c) Microsoft. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#
+
+set -e
+
+script_dir=$(cd "$(dirname "$0")" && pwd)
+build_root=$(cd "${script_dir}/.." && pwd)
+build_folder=$build_root"/build"
+
+CORES=$(grep -c ^processor /proc/cpuinfo 2>/dev/null || sysctl -n hw.ncpu)
+
+# Java binding
+pushd $build_root/bindings/java/gateway-java-binding
+mvn clean install
+popd
+
+rm -rf $build_folder
+mkdir -p $build_folder
```

Comment:
Yes, `set -e` causes the script to exit if any command fails. `set -e` has it's caveats--it's not a panacea. But I think it works fine for this script so far (e.g., IIRC the may be problems for commands in `if` statements, but we don't have any here yet).